compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -317,6 +317,58 @@ fn do_normalize_predicates<'tcx>(
     }
 }
 
+// A temporary hack. Do not use this.
+// See `deeply_normalize_param_env_or_error`.
+#[instrument(level = "debug", skip(tcx, elaborated_env))]
+fn do_deeply_normalize_predicates<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    cause: ObligationCause<'tcx>,
+    elaborated_env: ty::ParamEnv<'tcx>,
+    predicates: Vec<ty::Clause<'tcx>>,
+) -> Result<Vec<ty::Clause<'tcx>>, ErrorGuaranteed> {
+    let span = cause.span;
+    let infcx = tcx
+        .infer_ctxt()
+        .with_next_trait_solver(true)
+        .ignoring_regions()
+        .build(TypingMode::non_body_analysis());
+    let predicates = match crate::solve::deeply_normalize::<_, FulfillmentError<'tcx>>(
+        infcx.at(&cause, elaborated_env),
+        predicates,
+    ) {
+        Ok(predicates) => predicates,
+        Err(errors) => {
+            let reported = infcx.err_ctxt().report_fulfillment_errors(errors);
+            return Err(reported);
+        }
+    };
+
+    debug!("do_normalize_predicates: normalized predicates = {:?}", predicates);
+
+    // The next solver doesn't ignore region constraints so we do it manually.
+    drop(infcx.take_registered_region_obligations());
+    drop(infcx.take_registered_region_assumptions());
+    drop(infcx.take_and_reset_region_constraints());
```

Comment:
hmm, this feels... really bad xd

like, that's kind of the point of what we're doing here, but still

Does not dropping these things and instead not emitting a `span_delayed_bug` if there are errors also work? And also add an explicit comment referencing the FIXME from the function docs